### ISSUE TITLE
Local Multiprocessing

### ISSUE THREAD

**Issue by crantila** [ELVIS-Project/vis-framework#296]
Use the standard library's "multiprocessing" module whenever possible. At minimum, this should mean parallel importing in ``WorkflowManager.load()``, and parallel analysis of voice combinations in ``Indexer.do_multiprocessing()``.

**Comment by crantila** [comment]
@alexandermorgan Seems like you've effectively finished this already?

**Comment by alexandermorgan** [comment]
Yes but I only did multiprocessing of series indexers, and even then only the ones that call the do_multiprocessing method in indexer.py. So at present there is no multiprocessing at the workflowmanager piece level. I also added an on/off switch for multiprocessing for series indexers that are capable of using multiprocessing. Note that its default is on though (True). Myke has recently provided some advice on how to multiprocess with music21 which seems like it should make it possible to multiprocess streams which would allow for voice-level multiprocessing of stream indexers and also piece-level multiprocessing. While this would hopefully not be too much work to do, it is not a priority for me at this stage. Here are some links to helpful advice/documentation Myke has provided about music21 that are pertinent to multiprocessing in VIS:
http://music21-mit.blogspot.ca/2015/06/parallel-computing-with-music21.html
This stack overflow question has comments by Myke at the bottom. Note that we used to use the freeze/thaw method Myke mentions but we later decided it wasn't worth it:
http://stackoverflow.com/questions/30133898/multiprocess-tasks-in-python-that-cant-be-pickled

**Comment by crantila** [comment]
So it seems like you've finished the easy part of ``Indexer.do_multiprocessing()`` but pieces are still loaded serially. If it's all the same to you, we might as well just go ahead and close this issue. If someone wants pieces to import faster, you can think about it then.

(BTW the point here is to help you clean out the garbage I left in the issue tracker...)

**Comment by ahankinson** [comment]
Rodan will load pieces in parallel, which I imagine we'll be using for any large workloads.

**Comment by crantila** [comment]
That's not really local, but it definitely solves the problem. Thanks!

**Comment by alexandermorgan** [comment]
FYI @affogarty
